samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <zephyr.h>
+#include <net/nrf_cloud.h>
+#include <shell/shell.h>
+#include "mosh_print.h"
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <net/nrf_cloud_pgps.h>
+#endif
+
+BUILD_ASSERT(
+	IS_ENABLED(CONFIG_NRF_CLOUD_MQTT) &&
+	IS_ENABLED(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD));
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+static struct k_work notify_pgps_work;
+#endif
+
+static const char cloud_shell_cmd_usage_str[] =
+	"MQTT connection to nRFCloud\n"
+	"Usage: cloud <subcommand>\n"
+	"\n"
+	"Subcommands:\n"
+	"  connect:      Establish MQTT connection to nRFCloud.\n"
+	"  disconnect:   Disconnect from nRFCloud.\n";
+
+static void cloud_shell_print_usage(const struct shell *shell)
+{
+	mosh_print_no_format(cloud_shell_cmd_usage_str);
+}
+
+#if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS)
+static int send_service_info(void)
```

Comment:
I'm not sure if it's really necessary to update the device shadow from modem_shell. And why it's tied to A-GPS/P-GPS configuration?

My guess is that it isn't necessary to update the shadow, because modem_shell doesn't send the location to the cloud anyway. It looks like this tells the cloud that the cloud UI should display the GPS location window. So, because we don't send the location to the cloud, this shouldn't be necessary. And if we still want to do that, then it probably shouldn't depend on the A-GPS/P-GPS flags but should be done always.

Reply:
You are right. This doesn't seem to do anything at the moment. I'll remove it.